### ISSUE TITLE
Top_P,Top_K and Min_P arguments for ollama models

### ISSUE THREAD

**Issue by athmanar** [microsoft/autogen#3873]
Hello,
I don't understand how to set the parameters for ollama models.
This is my configuration:

```
config_list= [{"model": "llama3:70b-instruct", "base_url": "http://localhost:11434/v1", 'api_key': 'ollama', "price": [0, 0]}]
code_creator = autogen.AssistantAgent( name="code_creator", llm_config={ "cache_seed": 43, "config_list": config_list, "temperature": 0.9, }) #WORKS!
code_creator = autogen.AssistantAgent( name="code_creator", llm_config={ "cache_seed": 43, "config_list": config_list, "temperature": 0.9, "top_k": 40,"min_p": 0.05,"top_p": 0.95}) #DOES NOT WORK!
code_creator = autogen.AssistantAgent( name="code_creator", llm_config={ "cache_seed": 43, "config_list": config_list, "temperature": 0.9,"top_p": 0.95}) #WORKS!
```
It seems only top_p is supported? Am i missing something?

**Comment by ekzhu** [comment]
Please read the documentation on Ollama usage: https://microsoft.github.io/autogen/0.2/docs/topics/non-openai-models/local-ollama#api-parameters

**Comment by athmanar** [comment]
@ekzhu  Actually it is not is working ..
```
            config_list= [{"model": "llama3:70b-instruct", "client_host": ollamakey, 'api_type': 'ollama',"temperature": 1,"top_k": 50,} ]
            autogen.AssistantAgent(name=name,llm_config={"cache_seed": cache_seed,"config_list": config_list,},system_message=system_message)
```

I think there is an implementation error:
```
    raise RuntimeError(f"Ollama exception occurred: {e}")
RuntimeError: Ollama exception occurred: Client.chat() got an unexpected keyword argument 'temperature'
```

**Comment by ekzhu** [comment]
@marklysze is this something you can take a look at?

**Comment by athmanar** [comment]
@ekzhu @marklysze One fix could be adding the temperature,top_k etcg in the options dictionary.

```
options={'temperature':temperature,'top_k':top_k}
config_list[0]['options']=options
```
This does not create any issues but I am not sure if ollama is honoring these options or simple ignoring.

**Comment by marklysze** [comment]
@athmanar, would you be able to provide a simple code example with agents and messages and I'll test it

**Comment by marklysze** [comment]
Thanks, let me try it out (though can't run 70b so will run a smaller model ;) )

**Comment by marklysze** [comment]
@athmanar, please see PR 3952, you can try replacing your ollama.py file and see if it fixes it for you. It worked for me with your code.

**Comment by marklysze** [comment]
Okay, so it looks I already corrected it 3 weeks ago #3593. So it's probably a version thing. So in this repo, this file is the latest one:
https://github.com/microsoft/autogen/blob/0.2/autogen/oai/ollama.py

If you're using the `autogen` package, [which is this repo](https://github.com/autogenhub/autogen/blob/main/autogen/oai/ollama.py), then that is also updated.

Can you try to update your package or replace your `ollama.py`.

**Comment by athmanar** [comment]
@marklysze I can confirm that this now works. I replaced only the ollama file.
